WolvenKit.Modkit/RED4/Tools/MeshTools.cs

```diff
@@ -581,9 +581,10 @@ public static List<RawMeshContainer> ContainRawMesh(MemoryStream gfs, MeshesInfo
 
                 for (var e = 0; e < apps.Count; e++)
                 {
-                    // Null-proof materials
-                    var d = info.appearances[apps[e]];
-                    info.appearances[apps[e]] = InitializeDefaultMaterial(info.meshCount, d.NotNull());
+                    var appearance = apps[e];
+                    // Null-proof materials by making sure we have one material instance per submesh
+                    var d = info.appearances[appearance];
+                    info.appearances[apps[e]] = InitializeDefaultMaterial(info.meshCount, ref d);
                     meshContainer.materialNames[e] = d[index];
```

Comment:
Why does `InitializeDefaultMaterial` now both return the value AND the same data as ref?
Isn't it the same as `meshContainer.materialNames[e] = info.appearances[apps[e]][index];` now?

Reply:
Now that you say it. I'll fix it

Reply:
Updated